### PR TITLE
Using cmake to compile libtorrent

### PR DESCRIPTION
**Pull request by devos50**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Also see https://github.com/Homebrew/homebrew-core/pull/43858

### CODE REVIEW

File: Formula/libtorrent-rasterbar.rb [hunk 1]
```diff
@@ -21,29 +21,36 @@ class LibtorrentRasterbar < Formula
...
-      --disable-dependency-tracking
-      --disable-silent-rules
-      --prefix=#{prefix}
-      --enable-encryption
-      --enable-python-binding
-      --with-boost=#{Formula["boost"].opt_prefix}
-      --with-boost-python=boost_python37-mt
-      PYTHON=python3
+    args = std_cmake_args + %W[
+      -DCMAKE_BUILD_TYPE=Release
```
Comment (fxcoudert) [inline Formula/libtorrent-rasterbar.rb]: This is already provided by `std_cmake_args`

File: Formula/libtorrent-rasterbar.rb [hunk 2]
```diff
@@ -21,29 +21,36 @@ class LibtorrentRasterbar < Formula
...
+      -Dpython-bindings=ON
+      -DCMAKE_CXX_STANDARD=11
+      -DCMAKE_INSTALL_PREFIX=#{prefix}
```
Comment (fxcoudert) [inline Formula/libtorrent-rasterbar.rb]: This is already provided by `std_cmake_args`

File: Formula/libtorrent-rasterbar.rb [hunk 3]
```diff
@@ -21,29 +21,36 @@ class LibtorrentRasterbar < Formula
   depends_on "pkg-config" => :build
   depends_on "boost"
   depends_on "boost-python3"
+  depends_on "cmake" => :build
```
Comment (fxcoudert) [inline Formula/libtorrent-rasterbar.rb]: Move it up, above `pkg-config`

**Comment by fxcoudert** [comment]
There are warnings:

```
CMake Warning at /usr/local/lib/cmake/boost_system-1.71.0/libboost_system-variant-shared.cmake:59 (message):
  Target Boost::system already has an imported location
  '/usr/local/lib/libboost_system-mt.dylib', which will be overwritten with
  '/usr/local/lib/libboost_system.dylib'
Call Stack (most recent call first):
  /usr/local/lib/cmake/boost_system-1.71.0/boost_system-config.cmake:43 (include)
  /usr/local/lib/cmake/Boost-1.71.0/BoostConfig.cmake:117 (find_package)
  /usr/local/lib/cmake/Boost-1.71.0/BoostConfig.cmake:182 (boost_find_component)
  /usr/local/Cellar/cmake/3.15.3/share/cmake/Modules/FindBoost.cmake:443 (find_package)
  cmake/Modules/LibtorrentMacros.cmake:43 (find_package)
  CMakeLists.txt:674 (find_public_dependency)
```

and errors:

```
CMake Error at /usr/local/lib/cmake/Boost-1.71.0/BoostConfig.cmake:117 (find_package):
  Could not find a package configuration file provided by "boost_python"
  (requested version 1.71.0) with any of the following names:

    boost_pythonConfig.cmake
    boost_python-config.cmake

  Add the installation prefix of "boost_python" to CMAKE_PREFIX_PATH or set
  "boost_python_DIR" to a directory containing one of the above files.  If
  "boost_python" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  /usr/local/lib/cmake/Boost-1.71.0/BoostConfig.cmake:182 (boost_find_component)
  /usr/local/Cellar/cmake/3.15.3/share/cmake/Modules/FindBoost.cmake:443 (find_package)
  bindings/python/CMakeLists.txt:56 (find_package)
```

**Comment by Noctem** [comment]
In #38382, the author (@arvidn) recommended using boost-build. I intended to submit a PR with that change, but haven't gotten around to it. I'm happy to let @devos50 take care of that if he'd like, otherwise I can give it a try in the next day or so.

**Comment by arvidn** [comment]
I’d say cmake is a pretty safe bet for building libtorrent too. Both vcpkg and brew rely on it. The only thing is that I’m not personally super familiar with it.

**Comment by devos50** [comment]
Sorry for the late response, I am currently a bit busy.

> In #38382, the author (@arvidn) recommended using boost-build. I intended to submit a PR with that change, but haven't gotten around to it. I'm happy to let @devos50 take care of that if he'd like, otherwise I can give it a try in the next day or so.

No problem, please go ahead 👍

**Comment by fxcoudert** [comment]
Closing for now…

**Pull request closed without merging**